### PR TITLE
Rajitha assignment2 recursive

### PR DESCRIPTION
**Pull request by ghost**
Rajitha assignment2 recursive

### CODE REVIEW

File: Rajitha/recursion-practice.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
+/****************************************************
+* Create a function to get the factoral of a number.
+* Input: factoral(5)  Output: 120
+****************************************************/
+let fact = 1;
```
Comment (koding-101) [inline Rajitha/recursion-practice.js]: don't use global scope to store variables. 

wrap this function in another function. - RY

File: Rajitha/recursion-practice.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+}
+console.log(numPower(2,5));
+
+/***************************************************
+* Reverse a string
+* Input: reverse("Recursion Sucks")
+* Output: "skcuS noisruceR"
+****************************************************/
+let revString = '';
+const  reverseString = str => {
```
Comment (koding-101) [inline Rajitha/recursion-practice.js:35]: please create different test/edge cases. Don't rely only on the example given. - RY

File: Rajitha/recursion-practice.js (hunk 1 above)
Comment (koding-101) [reply Rajitha/recursion-practice.js]: Agreed, the point of recursion is to have a self contained function that will loop itself and spit out the result. With a global variable, it is easier to just use a loop. Try doing these problems without a global variable. Instead have a variable inside the function itself that will contain the result.

Remember how I taught you guys 2-3 weeks ago where each time the function is called again, return the result to the previous function call?

File: Rajitha/recursion-practice.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
+/****************************************************
+* Create a function to get the factoral of a number.
+* Input: factoral(5)  Output: 120
+****************************************************/
+var factorial = function(number) {
+  if (number <= 0) {
+    return 1;
+  } else {
```
Comment (koding-101) [inline Rajitha/recursion-practice.js:8]: no issue,

for me, i prefer not to have the else only the if part is needed. -RY

File: Rajitha/recursion-practice.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    return (number * factorial(number - 1));
+  }
+};
+factorial(6);
+
```
Comment (koding-101) [inline Rajitha/recursion-practice.js:13]: please add more test cases - RY.

File: Rajitha/recursion-practice.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+* reason: 2x2x2x2x2
+**********************************************************/
+var numPower = function(num,exp) {
+  if (exp === 0){
+    return 1;
+  } else {
+    return num * numPower(num,exp-1);
+  }
+};
+numPower(2,5);
```
Comment (koding-101) [inline Rajitha/recursion-practice.js:28]: same as above.... thinking of other possible number combinations. 
-  numPower(100, 1).
-  numPower(....


RY

File: Rajitha/recursion-practice.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
+/****************************************************
+* Create a function to get the factoral of a number.
+* Input: factoral(5)  Output: 120
+****************************************************/
+var factorial = function(number) {
+  if (number <= 0) {
```
Comment (koding-101) [inline Rajitha/recursion-practice.js:6]: Actually I was wrong with my demo. This should be `===1` and another case should be added for 0. factoral(0) should equal 0 not 1

File: Rajitha/recursion-practice.js [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+/*********************************************************
+* Create a function to get the power of a number.
+* Multiply the first input x(second input) number of times
+* Input: power(2, 5)
+* Output: 32
+* reason: 2x2x2x2x2
+**********************************************************/
+var numPower = function(num,exp) {
+  if (exp === 0){
+    return 1;
```
Comment (koding-101) [inline Rajitha/recursion-practice.js:23]: This would be wrong if `numPower(0,0)`
